config/config.go

```diff
@@ -73,6 +73,33 @@ const (
 
 	SettingInventoryTimeout        = "inventory_timeout"
 	SettingInventoryTimeoutDefault = 10
+
+	// SettingPresignAlgorithm sets the algorithm used for signing
+	// downloadable URLs. This option is currently ignored.
+	SettingPresignAlgorithm        = "presign.algorithm"
+	SettingPresignAlgorithmDefault = "HMAC256"
+
+	// SettingPresignSecret sets the secret for generating signed url.
+	// For HMAC type of algorithms the value must be a base64 encoded
+	// secret. For public key signatures, the value must be a path to
+	// the private key (not yet supported).
+	SettingPresignSecret        = "presign.secret"
+	SettingPresignSecretDefault = "qF4/MZzQmHTJ+nrlr26b3g=="
```

Comment:
I don't think it's a good idea to provide a default here, as this can lead to security issues.
Can we find a better way to deal with it? A random value at start-up is good enough, IMHO.

Reply:
Good call!
Actually I discovered that I incorrectly configured the defaults below as well.
I pushed new commit generating making the secret default to a random base64 string of 32 bytes if not configured explicitly.

Reply:
+1, that was my comment from writing acceptance tests - the default wasn't picked up, and it was 'good' imo.